### ISSUE TITLE
Issues running on Cray with tcp Sockets tests

### ISSUE THREAD

**Issue by markwdalton** [JuliaParallel/MPI.jl#270]
The other issue running on the Cray was related to the tcp sockets test.
It was a bit hard to parse I had asked it to be re-run with aprun -T.

On the XC might not support irrelevant features, such as "sockets".   So either ignore the test or in the test look for the architecture.

 /home/willmore/.julia/packages/MPI/U5ujD/test/test_cman_tcp.jl 
* This will take a little more work, but re-running with -T (to sync stdout/stderr) may help. 

failed process: Process(`aprun -b -N1 -n 4 /location/julia/1.0.3/bin/julia --code-coverage=none /user/.julia/packages/MPI/U5ujD/test/test_cman_tcp.jl`, ProcessExited(137)) [137] 
* aprun command looks fine 

But the errors a bit confusing when they are interleaved. But it does appear to be in the Socket area. 

* aprun -T - helps with Syncing the stdout/stderr in the job may help: (Adding the -T) 

errors: 
ERROR: ERROR: ERROR: LoadError: LoadError: LoadError: IOError: connect: connection refused (ECONNREFUSED)IOError: connect: connection refused (ECONNREFUSED)IOError: connect: connection refused (ECONNREFUSED) 
Stacktrace: 
Stacktrace: 
Stacktrace: 
[1] 
[1] 
[1] try_yieldtotry_yieldtotry_yieldto(((::::::typeof(Base.typeof(Base.typeof(Base.ensure_rescheduled), ::Base.RefValue{Task}) at ./event.jl:196 
[2] wait() at ./event.jl:255 
[3] wait(::Condition) at ./event.jl:46ensure_rescheduled), 
[4] ::Base.RefValue{stream_wait(::Task}) at ./event.jl:196Sockets.TCPSocket 
[2] , ::wait()Condition) at ./stream.jl:47 at ./event.jl:255 
[5] wait_connected 
[3] wait((::Sockets.::ConditionTCPSocket) at ./stream.jl:259) at ./event.jl:46 

test_cman_tcp.jl: mgr = MPI.start_main_loop(MPI.TCP_TRANSPORT_ALL) 
-> cman.jl:start_main_loop 
MPI.Initialized() && MPI.Init()

**Comment by simonbyrne** [comment]
I'm not sure about this one?

**Comment by markwdalton** [comment]
I understand.  I think Julia could work around this without the use of
MPI/TCP.  But the
user wanted to use the MPI.   I was not sure if you had others running
Julia->MPI on various
Cray machines XE/XK, XC currently.  I am still wanting to get you access to
a Cray.

Mark
